File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v2/SpotOrderApi.java
package com.bitget.openapi.api.v2;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface SpotOrderApi {

    // order
    @POST("/api/v2/spot/trade/place-order")
    Call<ResponseResult> placeOrder(@Body Map<String, String> body);

    @POST("/api/v2/spot/trade/batch-orders")
    Call<ResponseResult> batchOrders(@Body Map<String, String> body);

    @POST("/api/v2/spot/trade/cancel-order")
    Call<ResponseResult> cancelOrder(@Body Map<String, String> body);

    @POST("/api/v2/spot/trade/batch-cancel-order")
    Call<ResponseResult> batchCancelOrder(@Body Map<String, String> body);

    @GET("/api/v2/spot/trade/unfilled-orders")
    Call<ResponseResult> unfilledOrders(@QueryMap Map<String, String> body);

    @GET("/api/v2/spot/trade/history-orders")
    Call<ResponseResult> historyOrders(@QueryMap Map<String, String> body);

    @GET("/api/v2/spot/trade/fills")
    Call<ResponseResult> fills(@QueryMap Map<String, String> body);


    // plan
    @POST("/api/v2/spot/trade/place-plan-order")
    Call<ResponseResult> placePlanOrder(@Body Map<String, String> body);

    @POST("/api/v2/spot/trade/cancel-plan-order")
    Call<ResponseResult> cancelPlanOrder(@Body Map<String, String> body);

    @GET("/api/v2/spot/trade/current-plan-order")
    Call<ResponseResult> currentPlanOrder(@QueryMap Map<String, String> body);

    @GET("/api/v2/spot/trade/history-plan-order")
    Call<ResponseResult> historyPlanOrder(@QueryMap Map<String, String> body);


    // trace
    @POST("/api/v2/copy/spot-trader/order-close-tracking")
    Call<ResponseResult> traderOrderCloseTracking(@Body Map<String, String> body);

    @GET("/api/v2/copy/spot-trader/order-current-track")
    Call<ResponseResult> traderOrderCurrentTrack(@QueryMap Map<String, String> body);

    @GET("/api/v2/copy/spot-trader/order-history-track")
    Call<ResponseResult> traderOrderHistoryTrack(@QueryMap Map<String, String> body);
}

File: bitget-node-sdk-api/src/lib/v2/SpotOrderApi.ts
import {BaseApi} from '../BaseApi';

export class SpotOrderApi extends BaseApi {

    placeOrder(qsOrBody: object) {
        const url = '/api/v2/spot/trade/place-order';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    batchOrders(qsOrBody: object) {
        const url = '/api/v2/spot/trade/batch-orders';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    cancelOrder(qsOrBody: object) {
        const url = '/api/v2/spot/trade/cancel-order';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    batchCancelOrder(qsOrBody: object) {
        const url = '/api/v2/spot/trade/batch-cancel-order';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    orderInfo(qsOrBody: object) {
        const url = '/api/v2/spot/trade/orderInfo';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    unfilledOrders(qsOrBody: object) {
        const url = '/api/v2/spot/trade/unfilled-orders';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    historyOrders(qsOrBody: object) {
        const url = '/api/v2/spot/trade/history-orders';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    fills(qsOrBody: object) {
        const url = '/api/v2/spot/trade/fills';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    placePlanOrder(qsOrBody: object) {
        const url = '/api/v2/spot/trade/place-plan-order';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    cancelPlanOrder(qsOrBody: object) {
        const url = '/api/v2/spot/trade/cancel-plan-order';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    currentPlanOrder(qsOrBody: object) {
        const url = '/api/v2/spot/trade/current-plan-order';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    historyPlanOrder(qsOrBody: object) {
        const url = '/api/v2/spot/trade/history-plan-order';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    traderOrderCloseTracking(qsOrBody: object) {
        const url = '/api/v2/copy/spot-trader/order-close-tracking';
        const headers = this.signer('POST', url, qsOrBody)
        return this.axiosInstance.post(url, qsOrBody, {headers})
    }

    traderOrderCurrentTrack(qsOrBody: object) {
        const url = '/api/v2/copy/spot-trader/order-current-track';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }

    traderOrderHistoryTrack(qsOrBody: object) {
        const url = '/api/v2/copy/spot-trader/order-history-track';
        const headers = this.signer('GET', url, qsOrBody)
        return this.axiosInstance.get(url, {headers, params: qsOrBody})
    }
}